packages/components/src/components/tag/tag.scss

```
@@ -1,70 +1,152 @@
 @use "sass:map";
 @use "@db-ui/foundations/build/scss/variables";
-@use "@db-ui/foundations/build/scss/color-placeholder";
 @use "@db-ui/foundations/build/scss/helpers/component";
 @use "@db-ui/foundations/build/scss/helpers/font";
+@use "@db-ui/foundations/build/scss/icon/icon-helpers";
+@use "@db-ui/foundations/build/scss/helpers/functions";
+@use "@db-ui/foundations/build/scss/color/color-variants";
 @use "../../styles/button-components";
 @use "../../styles/tag-components";
-@use "@db-ui/foundations/build/scss/helpers/functions";
+@use "../../styles/icon-passing";
 
-.db-tag {
+%no-text {
+	&[data-no-text="true"] {
+		--icon-margin-after: 0;
+		@include icon-helpers.is-icon-text-replace();
+		padding: 0;
+
+		label,
+		span {
+			@include icon-helpers.is-icon-text-replace();
+		}
+	}
+
+	&:not(&[data-no-text="true"]) {
```

```suggestion
	&:not([data-no-text="true"]) {
```

Is there a reason to include the `db-tag` class in there again ?